#159274730 packages and files to integrate hound

tutugodfrey opened the pull request:
#### What does this PR do?
- add configuration for HoundCI to report on style guide violations

#### Description of Task to be completed?
- install dependencies to integrate HoundCI
- setup configuration files

#### How should this be manually tested?
- Hound reports when a pull request is raised

#### What are the relevant pivotal tracker stories?
- #159274730

#### Screenshots (if appropriate)
- N/A

#### Questions:
- N/A

rhenshaw56 commented:
looks good but you got conflicts

rhenshaw56 commented:
resolve your conflicts and lets get this in

rhenshaw56 commented on package.json:
```diff
@@ -5,7 +5,8 @@
     "test": "echo \"no test specified\" && exit 0",
-    "start": "node index.js"
+    "start": "node index.js",
+    "lint": "eslint index.js ./models/ ./routes/ ./config/"
```
just do `eslint .` and it'll lint everything, then create an ignore file and add files you don't plan to lint in it

megame24 committed:
chore(root): integrate houndci

- install dependencies for linting
- configure eslint and add linting script to package.json
- add .hound.yml file to integrate hound

[Finishes #159274730]
